### PR TITLE
Adding tools to map from chromosome positions to mRNA coding sequence…

### PR DESCRIPTION
**Pull request by andreasprlic**
… position.

The genome module now also depends on the Guava library. This suggests that we should probably merge the Genome and Sequencing modules, since there is less and less difference between the two.

### CODE REVIEW

**Comment by sbliven** [comment]
Can this be merged, or does it depend on #536?

**Comment by andreasprlic** [comment]
does not depend on #536

File: biojava-genome/src/main/java/org/biojava/nbio/genome/util/ChromosomeMappingTools.java [hunk 1]
```diff
@@ -0,0 +1,1394 @@
...
+                        System.out.println(codingLength + " | " + tmp);
+                        System.out.println(" -> found position in coding exon #" + (i + 1) + ":  " + format(chromPos) + " " + format(start + 1) + " " + format(tmp) + " " + cdsStart + " " + codingLength);
+                    }
+
+                    return codingLength + tmp ;
+                }
+                // full exon is coding
+                codingLength += (end - start);
+                if (debug)
+                    System.out.println("     Exon        :" + format(start) + " - " + format(end) + " | " + format(end - start) + " | " + codingLength + " | " + (codingLength % 3));
```
Comment (josemduarte) [inline biojava-genome/src/main/java/org/biojava/nbio/genome/util/ChromosomeMappingTools.java]: All these should be done with `logger.debug` and not use a `debug` constant at all.

File: biojava-genome/src/main/java/org/biojava/nbio/genome/util/ChromosomeMappingTools.java [hunk 2]
```diff
@@ -0,0 +1,1394 @@
+package org.biojava.nbio.genome.util;
+
+
+import com.google.common.collect.Range;
+
+import org.biojava.nbio.core.sequence.ProteinSequence;
```
Comment (josemduarte) [inline biojava-genome/src/main/java/org/biojava/nbio/genome/util/ChromosomeMappingTools.java]: This import is not used

File: biojava-genome/src/main/java/org/biojava/nbio/genome/util/ChromosomeMappingTools.java [hunk 3]
```diff
@@ -0,0 +1,1394 @@
...
+
+    public static String formatExonStructure(GeneChromosomePosition chromosomePosition ){
+        if ( chromosomePosition.getOrientation() == '+')
+            return formatExonStructureForward(chromosomePosition);
+
+        return formatExonStructureReverse(chromosomePosition);
+
+    }
+
+    public static String printHTMLExonStructure(GeneChromosomePosition chromosomePosition ){
```
Comment (josemduarte) [inline biojava-genome/src/main/java/org/biojava/nbio/genome/util/ChromosomeMappingTools.java]: Any documentation for this? What kind of html output does this produce? Perhaps we shouldn't have this specific output thing here in biojava but elsewhere

File: biojava-genome/src/main/java/org/biojava/nbio/genome/util/ChromosomeMappingTools.java [hunk 4]
```diff
@@ -0,0 +1,1394 @@
...
+public class ChromosomeMappingTools {
+
+    public static final boolean debug = false;
+    private static final String newline = System.getProperty("line.separator");
+
+
+    public static final String CHROMOSOME = "CHROMOSOME";
+    public static final String CDS = "CDS";
+
+    public static String formatExonStructure(GeneChromosomePosition chromosomePosition ){
```
Comment (josemduarte) [inline biojava-genome/src/main/java/org/biojava/nbio/genome/util/ChromosomeMappingTools.java]: Any docs for this? What kind of formatting does this produce?

File: biojava-genome/src/main/java/org/biojava/nbio/genome/util/ChromosomeMappingTools.java [hunk 5]
```diff
@@ -0,0 +1,1394 @@
...
+    }
+
+
+
+    /** Extracts the exon boundaries in CDS coordinates. (needs to be divided by 3 to get AA positions)
+     *
+     * @param chromPos
+     * @return
+     */
+    public static List<Range> getCDSExonRanges(GeneChromosomePosition chromPos){
```
Comment (josemduarte) [inline biojava-genome/src/main/java/org/biojava/nbio/genome/util/ChromosomeMappingTools.java]: Range should be parameterised, the compiler gives warnings for it. Same for all other occurrences below

Comment (andreasprlic) [inline biojava-genome/src/main/java/org/biojava/nbio/genome/util/ChromosomeMappingTools.java]: Interesting! IntelliJ did not show this warning. Will fix...

**Comment by andreasprlic** [comment]
this addresses all of @josemduarte 's comments.

**Comment by josemduarte** [comment]
Thanks!

**Pull request merged**